### PR TITLE
Enable testifylint to lint test files, and mechanically fix reported issues

### PR DESCRIPTION
**Pull request by giorio94**
Enable testifylint to lint the usages of github.com/stretchr/testify, in an effort to improve the readability of tests (and failure outputs) and reduce the chance of flakes. This also aims to remove the need for mechanical code review comments to encourage the usage of idiomatic testify primitives.
    
The current configuration disables the "float-compare", "go-require" and "require-error" checkers, as extra effort is required to fix the issues reported by them (require-error appears to also lead to false positives when used inside EventuallyWithT blocks).

Note to the reviewers, all except the first five commits, and the last one, have been mechanically generated with the following script:

```bash
#!/bin/bash

while read -r LINTER; do

[[ ${LINTER} =~ ^# ]] && continue

# Required to prevent error-nil from attempting to apply the same fix twice to
# a single code-block, failing due to conflicts.
if [[ ${LINTER} = "error-nil" ]]; then
    find . -iname '*_test.go' | \
        xargs sed -i -e 's/require.Nil(t, testutils.WaitUntil/require.NoError(t, testutils.WaitUntil/'
fi

testifylint --fix --disable-all --enable ${LINTER} ./...
git status -s | awk '{ print $2}' | xargs -r goimports -w

git add .
git commit --signoff -F- <<EOF
treewide(tests): testifylint "${LINTER}" fixes

Generated by:

    testifylint --fix --disable-all --enable ${LINTER} ./...
EOF

done <<-EOF
error-nil
bool-compare
compares
contains
empty
error-is-as
expected-actual
formatter
len
negative-positive
nil-compare
regexp
suite-broken-parallel
suite-dont-use-pkg
suite-extra-assert-call
suite-thelper
EOF
```

### CODE REVIEW

**Review by christarazi — APPROVED** [review]

**Review by jschwinger233 — APPROVED** [review]

File: pkg/hubble/metrics/api/context_test.go [hunk 1]
```diff
@@ -514,26 +514,24 @@ func TestParseGetLabelValues(t *testing.T) {
 		TrafficDirection: pb.TrafficDirection_INGRESS,
 	}
 	assert.EqualValues(t,
-		mustGetLabelValues(opts, flow),
 		[]string{
-			// source_ip, source_pod, source_namespace, source_workload, source_workload_kind , source_app
```
Comment (rolinh) [inline pkg/hubble/metrics/api/context_test.go:519]: I think these comments were somewhat useful given it's just a slice of strings

**Review by rolinh — APPROVED** [review]

**Review by marseel — APPROVED** [review]

**Review by rastislavs — APPROVED** [review]

**Review by gandro — APPROVED** [review]
Looks good for my codeowners (besides what Robin pointed out in one of the Hubble files)! Thanks

File: pkg/hubble/metrics/api/context_test.go (hunk 1 above)
Comment (giorio94) [reply pkg/hubble/metrics/api/context_test.go:519]: Sure, I'll restore them (it seems that `--fix` is not smart enough to preserve the comments :disappointed:.

**Review by pippolo84 — APPROVED** [review]
Thanks!

**Review by rgo3 — APPROVED** [review]

**Review by ysksuzuki — APPROVED** [review]

**Review by derailed — APPROVED** [review]
@giorio94 Nice work Marco!

**Review by tommyp1ckles — APPROVED** [review]

**Comment by tommyp1ckles** [comment]
(Approving for Azure)

**Review by sayboras — APPROVED** [review]
Thanks a lot, a lot of linting error is probably due to my mistake.

**Comment by giorio94** [comment]
> Thanks a lot, a lot of linting error is probably due to my mistake.

Nothing to worry about, you did a great job at converting all tests to testify. I wasn't honestly aware of this linter at that time, otherwise it could have probably proved valuable to standardize the tests style.

**Review by nathanjsweet — APPROVED** [review]

**Comment by giorio94** [comment]
Rebased onto main, to fix an additional test that got merged in the meanwhile. Additionally, manually fixed a broken test assertion via https://github.com/cilium/cilium/pull/35237/commits/d715fbced5212769cb0ff3a6872fee54b0919457.

**Comment by giorio94** [comment]
Rebased onto main to fix conflict.

**Comment by giorio94** [comment]
Conflicted once more, let's try again.

**Comment by julianwiedmann** [comment]
Note this needs a rebase.

**Comment by giorio94** [comment]
> Note this needs a rebase.

Yeah, this PR turned out being a conflicts magnet (not that I expected anything different). Let's try again.

**Pull request merged**